internal/controller/dnsrecord_controller.go

```diff
@@ -220,25 +221,34 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	if err != nil {
 		setDNSRecordCondition(dnsRecord, string(v1alpha1.ConditionTypeReady), metav1.ConditionFalse,
 			"DNSProviderError", fmt.Sprintf("The dns provider could not be loaded: %v", err))
-		return r.updateStatus(ctx, previous, dnsRecord, false, []string{}, err)
+		return r.updateStatus(ctx, previous, dnsRecord, nil, false, []string{}, err)
 	}
 
+	probes := &v1alpha1.DNSHealthCheckProbeList{}
```

Comment:
if we defined this early would it help with readability. IE jusst pass in the empty list of probes to updateStatus? as passing in nil is a bit obscure

Reply:
defined up top and passing it always into the update status func